### PR TITLE
Feature/cdsct 101 update queries indexes

### PR DESCRIPTION
**Pull request by cmarcus-nhs**
Indexes added to:

- Cases.supplierId
- EmsSupplier.supplierId
- CdssSupplier.supplierId

Wherever a 'findAll' is done i've replaced this with a 'findAllBySupplierId'
I've not changed any other repository as they are usually searching by case id or some other id.

I've assumed that all of the queries that take the supplier id will be populated by the story next sprint. For the purposes of this, there is no functional difference.

### CODE REVIEW

File: src/main/java/uk/nhs/ctp/security/JWTLoginFilter.java [hunk 1]
```diff
@@ -63,7 +63,8 @@ protected void successfulAuthentication(HttpServletRequest req, HttpServletRespo
 		UserEntity userEntity = userRepository.findByUsername(auth.getName());
 		List<CdssSupplier> cdssSuppliers;
 		if (userEntity.getRole().equals("ROLE_NHS") || userEntity.getRole().equals("ROLE_ADMIN")) {
-			cdssSuppliers = cdssSupplierRepository.findAll();
+			//TODO: CDSCT-139 (this code doesn't seem to be used as it gets overwritten straight away)
```
Comment (mhouston-nhs) [inline src/main/java/uk/nhs/ctp/security/JWTLoginFilter.java:66]: Was it supposed to be an 'else'?

Comment (cmarcus-nhs) [reply src/main/java/uk/nhs/ctp/security/JWTLoginFilter.java:66]: I don't know, i'm not sure how this JWT auth on the EMS is supposed to behave.

File: src/main/java/uk/nhs/ctp/repos/CaseRepository.java [hunk 2]
```diff
@@ -3,26 +3,28 @@
...
+public interface CaseRepository extends PartitionedRepository<Cases, Long> {
   
   @Query("SELECT NEW uk.nhs.ctp.service.search.AuditSearchResultDTO(c.id, c.firstName, c.lastName, c.createdDate) " +
           "FROM Cases c " +
           "WHERE (:from IS NULL OR createdDate > :from) " +
           "AND (:to IS NULL OR createdDate < :to) " +
           "AND (:includeClosed = TRUE OR closedDate IS NULL)" +
-          "AND (:includeIncomplete = TRUE OR triageComplete = TRUE)")
+          "AND (:includeIncomplete = TRUE OR triageComplete = TRUE)" +
+          "AND (:supplierId IS NULL OR supplierId = :supplierId)")
```
Comment (mhouston-nhs) [inline src/main/java/uk/nhs/ctp/repos/CaseRepository.java:21]: Does this need the IS NULL clause? I don't think we ever want passing null for the supplier ID to mean all records.

File: src/main/java/uk/nhs/ctp/security/JWTLoginFilter.java (hunk 1 above)
Comment (mhouston-nhs) [reply src/main/java/uk/nhs/ctp/security/JWTLoginFilter.java:66]: We need to do some more work on this - I've added CDSCT-166 to capture requirements around user accounts

**Pull request merged**